src/coordination/include/coordination/coordinator_instance.hpp

```diff
@@ -76,6 +77,9 @@ class CoordinatorInstance {
  private:
   template <ranges::forward_range R>
   auto GetMostUpToDateInstanceFromHistories(R &&alive_instances) -> std::optional<std::string> {
+    if (ranges::empty(alive_instances)) {
```

Comment:
Maybe the problem happens when we pass alive_instances in the function... Don't know, seems strange. This is nice solution so let's leave it like this